base/arrayshow.jl

```diff
@@ -340,6 +340,9 @@ function show(io::IO, ::MIME"text/plain", X::AbstractArray)
     # checking for current :typeinfo (this could be changed in the future)
     io = IOContext(io, :typeinfo => eltype(X))
 
+    # Limit output if not specified
+    io = IOContext(io, :limit => get(io, :limit, true))
```

Comment:
I agree, it would be better not to change the default at this point.

Reply:
> show(stdout, "text/plain", array) can be useful to have all the values printed.

Why couldn't you use `show(stdout, array)` for that? The reasoning here is that if you want all value use `show(stdout, array)` and if you want something with a reasonable default use `show(stdout, "text/plain", array)`.

I guess we could also just call `display` instead of `repr` when converting the error to a string but that seems like abuse of `display`. Intuitively, I'd like that `display(x)` and `show(stdout, "text/plain", x)` were almost the same thing for plain text. The problem here might also be that it is so convoluted to control if the output should be limited or not.

Reply:
> Why couldn't you use show(stdout, array) for that?

It's just far less readable for big arrays. Aligned but not limited output is something I find useful, for which having to resort to an explicit `IOContext` call is too unconvenient IMHO.

I personally don't have a clear grasp of the overall logic of all the printing functions, so can't help much here. But I'm not convinced that having limited output by default is the right choice.

Reply:
It's useful when I have an array with sufficiently few columns to fit on the screen, and want to inspect the whole content. It's `Vector`s in the vast majority of my use-cases, but even then `show(x)` isn't ideal as the printing of some values are split over two lines (i.e. the line breaks are not inserted cleverly between two values; but this could probably be solved).

Reply:
A good example would be `x = collect(methods(eltype))` : `show(x)` is unreadable, so one item per line is preferable, and a truncated output is not so interesting.

Reply:
Notice that `Method` eltype is already handled separately and I don't intend to change that. There might be other cases to special case as well. It might also be that we should change the default to depend on the number of elements. Scrolling through 403 rows (like in `methods(*)`) is fine but probably not 10k and definitely not more than 100k rows.